From e7daeac0922d25b0064cb2ddfa7bb9a0f884eb0e Mon Sep 17 00:00:00 2001
From: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
Date: Tue, 28 Sep 2021 15:01:07 +0300
Subject: [PATCH 117/322] [FIO toup-squash] spl: minor fixes for
 FIT_SIGNATURE_STRICT mode
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This fixes a possible ambiguous 'else' [2] and using undeclared
function hang() [1].

[1]
common/spl/spl_fit.c: In function ‘spl_load_fit_image’:
common/spl/spl_fit.c:335:5: warning: implicit declaration of function ‘hang’ [-Wimplicit-function-declaration]
  335 |     hang();
      |     ^~~~
---------------------------------------------------------------------
[2]
common/spl/spl_fit.c:332:6: warning: suggest explicit braces to avoid ambiguous ‘else’ [-Wdangling-else]
  332 |   if (!fit_image_verify_with_data(fit, node, src, length))
      |      ^
---------------------------------------------------------------------
Fixes: dd9336eccb ("[FIO fromlist] spl: Add CONFIG_SPL_FIT_SIGNATURE_STRICT")
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
---
 common/spl/spl_fit.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/common/spl/spl_fit.c b/common/spl/spl_fit.c
index 6c1a53135a1..d22ad97bd17 100644
--- a/common/spl/spl_fit.c
+++ b/common/spl/spl_fit.c
@@ -16,6 +16,7 @@
 #include <asm/cache.h>
 #include <asm/global_data.h>
 #include <linux/libfdt.h>
+#include <hang.h>
 
 DECLARE_GLOBAL_DATA_PTR;
 
@@ -329,13 +330,14 @@ static int spl_load_fit_image(struct spl_load_info *info, ulong sector,
 	if (CONFIG_IS_ENABLED(FIT_SIGNATURE)) {
 		printf("## Checking hash(es) for Image %s ... ",
 		       fit_get_name(fit, node, NULL));
-		if (!fit_image_verify_with_data(fit, node, src, length))
+		if (!fit_image_verify_with_data(fit, node, src, length)) {
 			if (CONFIG_IS_ENABLED(FIT_SIGNATURE_STRICT)) {
 				puts("Invalid FIT signature found in a required image.\n");
 				hang();
 			} else {
 				return -EPERM;
 			}
+		}
 		puts("OK\n");
 	}
 
-- 
2.34.1

